build.gradle

```diff
@@ -1,122 +1,29 @@
-import org.springframework.boot.gradle.plugin.SpringBootPlugin
-
-import java.time.Instant
-import java.time.LocalDate
-
 plugins {
-    id 'java'
+    id("uk.gov.justice.hmpps.gradle-spring-boot") version "0.4.0"
     id 'groovy'
-    id 'org.springframework.boot' version '2.2.7.RELEASE'
-    id 'io.spring.dependency-management' version '1.0.9.RELEASE'
-    id 'org.owasp.dependencycheck' version '5.3.2.1'
-    id 'com.github.ben-manes.versions' version '0.28.0'
-    id 'se.patrikerdes.use-latest-versions' version '0.2.13'
-}
-
-repositories {
-    mavenLocal()
-    mavenCentral()
-}
-
-configurations {
-    agentDeps
-    agentDeps.transitive = false
-}
-
-project.sourceCompatibility = 11
-project.targetCompatibility = 11
-
-dependencyCheck {
-    failBuildOnCVSS = 5
-    suppressionFiles = ['dependency-check-suppress-spring.xml']
-    format = 'ALL'
-    analyzers {
-        assemblyEnabled = false
-    }
-}
-
-def isNonStable = { String version ->
-    def stableKeyword = ['RELEASE', 'FINAL', 'GA'].any { it -> version.toUpperCase().contains(it) }
-    def regex = /^[0-9,.v-]+(-r)?$/
-    return !stableKeyword && !(version ==~ regex)
-}
-
-dependencyUpdates {
-    rejectVersionIf {
-        isNonStable(it.candidate.version) && !isNonStable(it.currentVersion)
-    }
 }
 
-group = 'uk.gov.justice.digital.hmpps'
 
-def today = Instant.now()
-def todaysDate = LocalDate.now().format('yyyy-MM-dd')
-version = System.getenv().containsKey("BUILD_NUMBER") ? System.getenv("BUILD_NUMBER") : todaysDate
-
-jar {
-    baseName = rootProject.name
-    manifest {
-        attributes 'Implementation-Title': rootProject.name, 'Implementation-Version': version
-    }
-}
-
-springBoot {
-    buildInfo {
-        properties {
-            artifact = rootProject.name
-            version = version
-            group = group
-            name = rootProject.name
-            time = today
-            additional = [
-                    by                   : System.properties['user.name'],
-                    operatingSystem      : "${System.properties['os.name']} (${System.properties['os.version']})",
-                    machine              : InetAddress.localHost.hostName
-            ]
-        }
-    }
-}
-
-dependencyManagement {
-    imports {
-        mavenBom(SpringBootPlugin.BOM_COORDINATES)
-    }
-}
+ext["spring-security.version"] = "5.3.2.RELEASE"
 
 dependencies {
-    annotationProcessor "org.springframework.boot:spring-boot-configuration-processor"
```

Comment:
I don't think this has been added to the Gradle plugin - though I see it everywhere so there is a good argument to do that - so is this genuinely not needed by this project?

Reply:
I assume not given the app still starts and the int. tests work.